docs/tutorials/zarr-v3.md

```diff
@@ -0,0 +1,106 @@
+Please note that this introduces consistency issues – if you update the structure of the underlying `zarr` store i.e., remove a column from `obs`, the consolidated metadata will no longer be valid.
+Further, note that without consolidated metadata, we cannot guarantee your stored `AnnData` object will be fully readable.
+And even if it is fully readable, it will almost certainly be much slower to read.
+
+There are two ways of opening remote [`zarr` stores] from the `zarr-python` package, `fsspec` and `obstore`, and both can be used with `read_lazy`.
+[`obstore` claims] to be more performant out-of-the-box, but notes that this claim has not been benchmarked with the `uvloop` event loop, which itself claims to be 2× more performant than the default event loop for `python`.
+
+## Local data
+
+Local data generally poses a different set of challenges.
+First, write speeds can be somewhat slow and second, the creation of many small files on a file system can slow down a filesystem.
+For the "many small files" problem, `zarr` has introduced [sharding] in the v3 file format.
+Sharding requires knowledge of the array element you are writing (such as shape or data type), though, and therefore you will need to use {func}`anndata.experimental.write_dispatched` to use sharding.
+For example, you cannot shard a 1D array with `shard` sizes `(256, 256)`.
+Here is a short example, although you should tune the sizes to your own use-case and also use the compression that makes the most sense for you:
+
+```python
+import zarr
+import anndata as ad
+from collections.abc import Mapping
+from typing import Any
+
+ad.settings.zarr_write_format = 3 # Absolutely crucial! Sharding is only for the v3 file format!
+
+def write_sharded(group: zarr.Group, adata: ad.AnnData):
+    def callback(
+        func: ad.experimental.Write,
+        g: zarr.Group,
+        k: str,
+        elem: ad.typing.RWAble,
+        dataset_kwargs: Mapping[str, Any],
+        iospec: ad.experimental.IOSpec,
+    ):
+        if iospec.encoding_type in {"array"}:
+            dataset_kwargs = {
+                "shards": tuple(int(2 ** (16 / len(elem.shape))) for _ in elem.shape),
+                **dataset_kwargs,
+            }
+            dataset_kwargs["chunks"] = tuple(i // 2 for i in dataset_kwargs["shards"])
+        elif iospec.encoding_type in {"csr_matrix", "csc_matrix"}:
+            dataset_kwargs = {"shards": (2**16,), "chunks": (2**8,), **dataset_kwargs}
+        func(g, k, elem, dataset_kwargs=dataset_kwargs)
+
+    return ad.experimental.write_dispatched(group, "/", adata, callback=callback)
+```
+
+However, `zarr-python` can be slow with sharding throughput as well as writing throughput.
+Thus if you wish to speed up either writing, sharding, or both (or receive a modest speed-boost for reading), a bridge to the `zarr` implementation in Rust: https://zarrs-python.readthedocs.io/en/latest/ can help with that (see https://github.com/LDeakin/zarr_benchmarks for benchmarks):
```

Comment:
these are also still blank URLs. The first one should be linked via intersphinx, the second one should probably be the only regular markdown link in the document:

```md
… [`LDeakin/zarr_benchmarks` on GitHub][zarr-benchmarks] …

[zarr-benchmarks]: https://github.com/LDeakin/zarr_benchmarks
```